### PR TITLE
[RHPAM-2735] Add JVM configuration tests to BA Operator

### PR DESCRIPTION
**Pull request by desmax74**
Test on jvm conf in Crd
@spolti @tchughesiv @jakubschwan @bmozaffa

### CODE REVIEW

File: pkg/controller/kieapp/defaults/defaults_test.go [hunk 1]
```diff
@@ -4381,3 +4382,83 @@ func TestMergeDBDeployment(t *testing.T) {
 		})
 	}
 }
+
+func TestJvmCrd(t *testing.T) {
```
Comment (tchughesiv) [inline pkg/controller/kieapp/defaults/defaults_test.go:4386]: take a look at these crd schema tests in this file... maybe you can instead replicate what we already do here to accomplish the same thing for jvm specific crd validation?
https://github.com/kiegroup/kie-cloud-operator/blob/d6954019fcf87b0e2c452a5fc842b22de57a4850/pkg/controller/kieapp/test/crd_validation_test.go

File: pkg/controller/kieapp/defaults/defaults_test.go [hunk 2]
```diff
@@ -18,6 +18,7 @@ import (
 	appsv1 "github.com/openshift/api/apps/v1"
 	buildv1 "github.com/openshift/api/build/v1"
 	"github.com/stretchr/testify/assert"
+	"github.com/thedevsaddam/gojsonq/v2"
```
[the remark this answers is not in the document]
Comment (desmax74) [reply pkg/controller/kieapp/defaults/defaults_test.go:21]: I'll try

File: pkg/controller/kieapp/defaults/defaults_test.go (hunk 1 above)
Comment (desmax74) [reply pkg/controller/kieapp/defaults/defaults_test.go:4386]: Seems similar but the purpose of this test is to fail if the expected jvm params are removed from crd, if someone change the yaml has to change the test to alter the params

Comment (desmax74) [reply pkg/controller/kieapp/defaults/defaults_test.go:4386]: Ok

**Comment by tchughesiv** [comment]
@desmax74 looks great, just squash the commits and we'll get it merged.

**Comment by desmax74** [comment]
Why ? GitHub squash the commits automatically, almost from 2017

**Comment by tchughesiv** [comment]
@desmax74 are you sure? looking at one of your previous PR's, the commits weren't squashed and can still be seen in commit history.
https://github.com/kiegroup/kie-cloud-operator/pull/428

**Comment by desmax74** [comment]
@tchughesiv https://github.blog/2016-04-01-squash-your-commits/

**Comment by tchughesiv** [comment]
@desmax74 we try not to manually merge our PRs... openshift-ci system handles for us if tests pass and acks are in place.

**Pull request merged**